contrib/cirrus/cache_release.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+set -e
+
+source $(dirname $0)/lib.sh
+
+req_env_var GOSRC CIRRUS_BUILD_ID
+
+cd $GOSRC
+
+make release_info.txt
+FIELDS="RELEASE_BASENAME RELEASE_VERSION RELEASE_DIST RELEASE_DIST_VER RELEASE_ARCH RELEASE_FILE"
+read "$FIELDS" < release_info.txt
+req_env_var $FIELDS
+
+if [[ "$CI" == "true" ]]
+then
+    req_env_var CIRRUS_BRANCH CIRRUS_TASK_NAME CIRRUS_PR
+
+    # Clearly identify source PR or branch
+    if [[ "$CIRRUS_BRANCH" != "master" ]]
+    then
+        NAME_SUFFIX="pr$CIRRUS_PR"
+    else
+        NAME_SUFFIX="$CIRRUS_BRANCH"
+    fi
+
+    # Same script re-used for both uploading and downloading to avoid duplication
+    if [[ "$(basename $0)" == "cache_release.sh" ]]
+    then # Upload archive of tested content for downloading by release task
+        # Drop version to enable "latest" representation
+        ARCHIVE_NAME="${RELEASE_BASENAME}-${NAME_SUFFIX}-${RELEASE_DIST}-${RELEASE_DIST_VER}-${RELEASE_ARCH}.tar.bz2"
+        mv "$RELEASE_FILE" "$ARCHIVE_NAME"
+        # ref: https://cirrus-ci.org/guide/writing-tasks/#http-cache
+        curl -s -X POST --data-binary=@$ARCHIVE_NAME http://$CIRRUS_HTTP_CACHE_HOST/$ARCHIVE_NAME
+
+    elif [[ "$(basename $0)" == "release_cache.sh" ]]
+    then # Collect all archives for Cirrus-CI to upload
+        req_env_var RELEASES
+        # ref: https://cirrus-ci.org/guide/writing-tasks/#artifacts-instruction
+        cd /  # Source path becomes part of the URL
+        echo "$RELEASES" | while read REL_NAME
```

Comment:
The only definition I see of `RELEASES` is in .cirrus.yml and it's a one-liner, not a newline-separated list. Therefore I suspect this will not work. What you probably want to do instead is:
```bash
for REL_NAME in $RELEASES; do
```
(do not quote `$RELEASES`. Leave it exactly as written above).

Reply:
k, got it.

Reply:
I still don't see this fixed: in my latest checked-out PR, I see you removed the quotes but still do 'echo | while`. That will not work. Can you try exactly as I spelled it out above?

Reply:
Simplified / replaced by `for release_archive_name in $(ls -1 *.zip)...` as part of echo out a expected URL.